src/prefect/_internal/compatibility/pydantic/__init__.py

```diff
@@ -0,0 +1,4 @@
+try:
+    from pydantic.v1 import *
```

Comment:
Amazing! While you're in here, can you also revert the pin in https://github.com/PrefectHQ/prefect/blob/1d611266959de2f146dfc05a5e3e402a7e3cd1bb/requirements.txt#L27 (i.e. revert #10144)

Do you think we should have run our test suite with both Pydantic v1 and v2? How complete do we need our test matrix to be? (i.e. do we need Python [3.8, 3.9, 3.10, 3.11] x Pydantic [1, 2]) or can we take a shortcut to avoid doubling our test runtime again?

Reply:
Ah, nevermind, just read your PR description!

Reply:
I'll revert the pin once I've got tests passing on v1 👍 

I personally think testing against v2 is sufficient, but open to pushback on that - our tests already take so long and have non-deterministic errors, so I don't want to compound that if we can avoid it

Reply:
> I personally think testing against v2 is sufficient

I think that makes sense, and if we learn otherwise then we can always add a small number of v1 tests (i.e. test with the latest version of Python + Pydantic v1).

It's unlikely that future patch versions of v1 will break anything given that upstream is focused on v2

> our tests already take so long and have non-deterministic errors, so I don't want to compound that if we can avoid it

100% agreed!